components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,955 @@
+module scream_scorpio_interface
+
+!==============================================================================!
+! This module handles the Fortran interface to the PIO library of input/output
+! subroutines.  The essential set of steps to enable and use PIO for creating
+! output are as follows:
+!                                  OUTPUT
+! Initialization:
+! 1) Gather the pio_subsystem and pio_iotype information for the EAM component
+! as assigned by the component coupler.
+!    This is accomplished during eam_init_pio by calling
+!    'eam_init_pio_subsystem'
+! 2) For each output file "create" a file in PIO and record the unique file
+! descriptor. 
```

Comment:
Eventually we will need to be able to open files outside of eam_init_pio (e.g. if we write a new file every timestep). I don't think you should make any changes in this PR, but worth noting.

Reply:
yes, definitely.  My plan is for eam_init_pio to eventually just be needed for gathering the pio_subsystem and other universal tasks and have C++ directly call the file creation routines.